### ISSUE TITLE
[Feature Request] Cloud Storage Support

### ISSUE THREAD

**Issue by timbremer** [UnevenSoftware/LeafPic#304]
It would be awesome if users could add their Google Drive / Dropbox account to LeafPic and it would display their images like normal ones. See QuickPic for reference.

**Comment by wyatt8740** [comment]
This would result in the 'Uses or Promotes non-free services' flag in F-Droid.
If something like this were to be implemented however I'd hope to also see support for WebDAV.

**Comment by Bugsbane** [comment]
Personally I would love to just see support for Webdav so Nextcloud/ownCloud etc were supported, however I would put this kind of functionality way behind the basics like being able to rotate an image ;)

**Comment by wyatt8740** [comment]
Agreed; I'd like webDAV so I can use my own Apache server as an image host, however.

**Comment by timbremer** [comment]
@wyatt8740 Dropbox and Google Drive are not primarily paid services.

**Comment by wyatt8740** [comment]
@timbremer I was using 'free' in the richard stallman sense. The source for the sites in question is not open, even if there are APIs to access them. Github, too, is like this.

The F-Droid folks flag apps that use such sites.

I recommend some reading if that didn't clear things up.

https://www.gnu.org/philosophy/open-source-misses-the-point.en.html

> This is a matter of freedom, not price, so think of “free speech,” not “free beer.”
